### PR TITLE
feat: Allow skipping Wave server TLS verification #1547

### LINKED ISSUE #1547: Ability to turn off SSL cert verification when doing requests from Wave server (e.g. Keycloak)
#### Is your feature request related to a problem? Please describe

When WAVE server is running with keycloak which uses self-signed certificates it returns the following error:
```
panic: failed connecting to OIDC provider: Get "https://192.168.4.38:8443/auth/realms/master/.well-known/openid-configuration": x509: cannot validate certificate for 192.168.4.38 because it doesn't contain any IP SANs

goroutine 1 [running]:
github.com/h2oai/wave.Run({{0x401f3d, 0x9}, {0x4013ba, 0x7}, {0x401048, 0x6}, {0x40074e, 0x1}, {0x10d2468, 0x17}, ...})
	/home/odisseas/wave/server.go:101 +0x1aa0
main.main()
	/home/odisseas/wave/cmd/wave/main.go:256 +0x166c
```

#### Describe the solution you'd like

It would be helpful if it would be possible to turn off SSL cert verification

### PR DESCRIPTION
**Pull request by mturoci**
Closes #1547

### CODE REVIEW

File: website/docs/configuration.md [hunk 1]
```diff
@@ -139,13 +141,22 @@ H2O_WAVE_PRIVATE_DIR [2]
...
 
 - [1] `1`, `t`, `true` to enable; `0`, `f`, `false` to disable (case insensitive).
 - [2] Use OS-specific path list separator to specify multiple arguments - `:` for Linux/OSX and `;` for Windows. For example, `H2O_WAVE_PUBLIC_DIR=/images/@./files/images:/downloads/@./files/downloads`.
 
+### TLS verification
+
+During development, you might want to test out TLS encryption, e.g. communication between Wave server and Keycloak. The easiest thing to do is to generate a self-signed certificate. However, Wave server verifies certificates for all communication by default, thus would throw an error for a self-signed one. ***FOR DEVELOPMENT PURPOSES ONLY***, it's possible to turn off the check using either `H2O_WAVE_VERIFY_TLS` environment variable or `verify-tls` parameter.
+
+:::warning
+Make sure the TLS verification is not turned off in production environments as it's a huge security hole.
```
Comment (lo5) [inline website/docs/configuration.md:157]:
```suggestion
**Disabling TLS verification is a security risk.** Make sure TLS is not disabled in production environments.
```

**Review by lo5 — CHANGES_REQUESTED** [review]
Thanks @mturoci - minor (refactoring) changes.

File: cmd/wave/main.go [hunk 2]
```diff
@@ -120,6 +122,7 @@ func main() {
 	stringVar(&rawAuthScopes, "oidc-scopes", "", "OIDC scopes, comma-separated (default \"openid,profile\")")
 	stringVar(&rawAuthURLParams, "oidc-auth-url-params", "", "additional URL parameters to pass during OIDC authorization, in the format \"key:value\", comma-separated, e.g. \"foo:bar,qux:42\"")
 	boolVar(&auth.SkipLogin, "oidc-skip-login", false, "do not display the login form during OIDC authorization")
+	boolVar(&verifyTLS, "verify-tls", true, "do not verify TLS certificates during external communication - DO NOT USE IN PRODUCTION")
```
[the remark this answers is not in the document]
Comment (mturoci) [reply cmd/wave/main.go:125]: Agree, done.

File: cmd/wave/main.go [hunk 3]
```diff
@@ -75,6 +76,7 @@ func main() {
 		httpHeadersFile      string
 		createAccessKey      bool
 		listAccessKeys       bool
+		verifyTLS            bool
```
[the remark this answers is not in the document]
Comment (mturoci) [reply cmd/wave/main.go:79]: Done.

File: cmd/wave/main.go [hunk 4]
```diff
@@ -251,6 +254,10 @@ func main() {
 		conf.Proxy = true // IDE won't function without proxy
 	}
 
+	if !verifyTLS {
```
[the remark this answers is not in the document]
Comment (mturoci) [reply cmd/wave/main.go:257]: Not necessarily. Skipping TLS verification can happen even if Wave itself is not using TLS certs. E.g. server is listening on HTTP, but wants to communicate with a Keycloak that is using a self-signed cert. Agree with colocating the TLS logic though, we should just not put it into the proposed if block IMO.

**Pull request merged**